pxd_fastpath.c

```diff
@@ -416,6 +416,45 @@ static void pxd_failover_initiate(struct pxd_device *pxd_dev, struct pxd_io_trac
 	queue_work(pxd_dev->fp.wq, &head->wi);
 }
 
+static int remap_io_status(int status)
+{
+	switch (status) {
+	case 0: // success
+	case -EOPNOTSUPP: // op not supported - no failover
+	case -ENOSPC: // no space on device - no failover
+	case -ENOMEM: // no memory - no failover
+		return status;
+	}
+
+	return -EIO;
+}
+
+// @head [in] - io head
+// @return - update reconciled error code
+static int reconcile_io_status(struct pxd_io_tracker *head)
+{
+	struct pxd_io_tracker *repl;
+	int status = 0;
+	int tmp;
+
+	BUG_ON(head->magic != PXD_IOT_MAGIC);
+	list_for_each_entry(repl, &head->replicas, item) {
+		BUG_ON(repl->magic != PXD_IOT_MAGIC);
+
+		tmp = remap_io_status(repl->status);
```

Comment:
A little confusing. If we are going to ignore anything != EIO, why call remap_io_status()?

Reply:
That's not true. This function consolidates response status from each replica and translates response status to -EIO to indicate that a failover if possible should be initiated. All non zero status indicates a failure. A few of them is not to be considered a failure scenario for initiating failover, those are passed along directly to the end user without initiating failover.